### PR TITLE
Add documents link to company details local nav

### PR DESCRIPTION
**Pull request by yeahfro**
This change will:
- add the `Documents` link to the company details local nav
- acceptance tests

screen shot 2017-11-21 at 13 26 11

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ const config = {
   mediumDateTimeFormat: 'D MMM YYYY, h:mma',
   paginationMaxResults: 10000,
   performanceDashboardsUrl: process.env.PERFORMANCE_DASHBOARDS_URL || 'https://mi.exportwins.service.trade.gov.uk',
-  omisArchivedDocumentsBaseUrl: process.env.OMIS_ARCHIVED_DOCUMENTS_BASE_URL,
+  archivedDocumentsBaseUrl: process.env.OMIS_ARCHIVED_DOCUMENTS_BASE_URL,
```
Comment (teneightfive) [inline config/index.js]: Are the OMIS documents and the company documents definitely going to be in the same location?

Comment (teneightfive) [reply config/index.js]: If so, should we rename the environment variable name too?

File: src/apps/companies/middleware/params.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ async function getCompany (req, res, next, id) {
 
     company.address = getHeadingAddress(company)
     res.locals.company = company
+    res.locals.archivedDocumentsUrlPath = company.archived_documents_url_path
```
Comment (teneightfive) [inline src/apps/companies/middleware/params.js]: Lots of `res.locals` repetition happening here now. Maybe use `assign`?

Comment (teneightfive) [reply src/apps/companies/middleware/params.js]: Could `archivedDocumentsUrlPath` not just be accessed from the `res.locals.company` property?

File: src/apps/middleware.js [hunk 3]
```diff
@@ -15,13 +17,23 @@ function setHomeBreadcrumb (name) {
...
   return function buildLocalNav (req, res, next) {
-    res.locals.localNavItems = items.map(item => {
+    const localNavItems = items.map(item => {
       const url = `${req.baseUrl}/${item.path}`
       return Object.assign(item, {
         url,
         isActive: res.locals.CURRENT_PATH === url,
       })
     })
+    if (res.locals.archivedDocumentsUrlPath) {
```
Comment (teneightfive) [inline src/apps/middleware.js]: Could you use `get(res.locals, 'company.archived_documents_url_path')`?

File: src/apps/middleware.js [hunk 4]
```diff
@@ -15,13 +17,23 @@ function setHomeBreadcrumb (name) {
...
+      localNavItems.push({
+        label: 'Documents',
+        target: '_blank',
+        url: `${config.archivedDocumentsBaseUrl}${res.locals.archivedDocumentsUrlPath}`,
```
Comment (teneightfive) [inline src/apps/middleware.js]: Could this also use `get(res.locals, 'company.archived_documents_url_path')`?

File: src/apps/companies/router.js [hunk 5]
```diff
@@ -82,7 +84,14 @@ router
 router.post('/:companyId/archive', archiveCompany)
 router.get('/:companyId/unarchive', unarchiveCompany)
 
-router.use('/:companyId', setLocalNav(LOCAL_NAV))
+router.use('/:companyId', (req, res, next) => {
+  LOCAL_NAV.push({
+    label: 'Documents',
+    url: config.archivedDocumentsBaseUrl + get(res.locals, `company.archived_documents_url_path`),
```
Comment (teneightfive) [inline src/apps/companies/router.js]: Don't think this needs backticks actually.

File: test/acceptance/features/companies/details.feature [hunk 6]
```diff
@@ -5,13 +5,32 @@ Feature: Company details
...
+      | Audit history               |
+      | Documents                   |
+    And the company details CDMS reference is displayed
 
   @companies-details--no-cdms-reference
   Scenario: Company does not have CDMS reference
 
     When browsing to company fixture Lambda plc
-    Then the company details CDMS reference is not displayed
+    Then there should be a local nav
```
Comment (feedmypixel) [inline test/acceptance/features/companies/details.feature]: nice 👍

File: test/unit/apps/omis/middleware.test.js [hunk 7]
```diff
@@ -28,7 +28,7 @@ describe('OMIS middleware', () => {
         getDitCompany: this.getDitCompanyStub,
       },
       '../../../config': {
-        omisArchivedDocumentsBaseUrl: omisArchivedDocumentsBaseUrl,
+        archivedDocumentsBaseUrl: archivedDocumentsBaseUrl,
```
Comment (teneightfive) [inline test/unit/apps/omis/middleware.test.js]: Just spotted, could this actually just used the shorthand reference?

**Comment by feedmypixel** [comment]
Of note there was a bug with the work `const url = item.url || `${req.baseUrl}/${item.path}`` that was causing all companies to go to the first company you selected. I have refactored this ti use the `isExternal` flag and updated tests accordingly.

I have also added documentation around the use of `ARCHIVED_DOCUMENTS_BASE_URL` env and also added this env to the circle config.

I have also rather cheekily merged it as it was approved - but I am sure @yeahfro and @teneightfive will be fine with that. If not I owe them 🍻

**Pull request merged**